src/main/scala/nl/knaw/dans/easy/stage/EasyStageDataset.scala

```diff
@@ -72,13 +73,22 @@ object EasyStageDataset {
     for {
       dataDir <- getDataDir
       _ <- mkdirSafe(s.sdoSetDir)
-      emdContent <- createDatasetSdo()
+      (emdContent,amdContent) <- createDatasetSdo()
       _ = log.info("Creating file and folder SDOs")
       _ <- createFileAndFolderSdos(dataDir, DATASET_SDO, emdContent.getEmdRights.getAccessCategory)
-    } yield ()
+    } yield (emdContent,amdContent)
   }
 
   def createFileAndFolderSdos(dir: File, parentSDO: String, rights: AccessCategory)(implicit s: Settings): Try[Unit] = {
+    val maybeSha1Map: Try[Map[String, String]] = Try {
+      val sha1File = "manifest-sha1.txt"
+      readFileToString(new File(s.bagitDir, sha1File))
+        .split("\\v+") // split into lines
+        .map(_.split("\\h+")) // split into tokens
+        .filter(_.nonEmpty) // skip empty lines
+        .map(a => if (a.length == 2 && !a(0).matches("[a-zA-Z0-9]")) a(1) -> a(0) else throw new Exception(s"Invalid line in $sha1File: ${a.mkString(" ")}"))
```

Comment:
Since you expect an `Array` with two elements here, you can do this with a pattern match inside the function as well. That's a little more readable IMO:

```scala
.map {
  case Array(x, y) if !x.matches("[a-zA-Z0-9]") => y -> x
  case array => throw new IllegalArgumentException(s"Invalid line in $sha1File: ${array.mkString(" ")}")
}
```

Note that you could also discard the second match, in which case the pattern match will automatically throw an exception, although then it is less clear where that exception comes from and you have less/no control over which type of exception you throw or what the message is... That's why I left this second case in.